src/java.base/share/classes/java/lang/invoke/DirectMethodHandle.java

```diff
@@ -621,10 +621,9 @@ Object checkCast(Object obj) {
     // Enumerate the different field kinds using Wrapper,
     // with an extra case added for checked references.
     static final int
-            FT_LAST_WRAPPER    = Wrapper.COUNT-1,
             FT_UNCHECKED_REF   = Wrapper.OBJECT.ordinal(),
-            FT_CHECKED_REF     = FT_LAST_WRAPPER+1,
-            FT_LIMIT           = FT_LAST_WRAPPER+2;
+            FT_CHECKED_REF     = Wrapper.VOID.ordinal(),
+            FT_LIMIT           = Wrapper.COUNT;
```

Comment:
I don't see why these changes are needed?

Reply:
These avoid wasting space in the field lambda form array.